src/core/sdl/render_sdl.cpp

```diff
@@ -400,6 +400,15 @@ void RenderSDL::loadTexture(StdPicture &target, uint32_t width, uint32_t height,
     m_textureBank.insert(texture);
 
     target.inited = true;
+    target.w = width;
+    target.h = height;
+    target.frame_w = width;
+    target.frame_h = height;
+
+    target.l.w_orig = width;
+    target.l.h_orig = height;
+    target.l.w_scale = 1.0f;
+    target.l.h_scale = 1.0f;
```

Comment:
When the original size is matched to the actual, the w_orig and h_orig must be zeroes. This logic is used to use a simplified coordinates formula for such cases in the rendering code. When you don't need to use the scaling feature, just set Orig width and height being zeroes.

Reply:
I see, so we should keep `w_orig` 0 unless it's a 1x texture. Then, if it's a 1x texture, we need to separately handle the cases where the original texture loader set `w_orig == 0` and `w_orig != 0`.

Reply:
Do you want me to fix this or will you?

Reply:
Interesting, well, as long as keeping them as defaults is a valid state, then let's not set them in loadTexture. But I do want the outcome following loadTexture to be a fully valid state (there should be no need to set anything externally if it's a 2x sprite, and if it's a 1x sprite, loadTexture_1x should have the proper outcome). Otherwise, it will be difficult to maintain everything in the future.

Reply:
Or, do you think we should move the scaling code into the load texture directly?

Reply:
I think it would be reasonable to move the scaling code into loadTexture, since it is an implementation detail the loader shouldn't need to think about. But, the signature of loadTexture shouldn't use FI_Bitmaps, and writing our own scaling function for non-Po2 scaling would be nontrivial.

Reply:
Maybe this is a feature we can do later, outside of this PR, whenever we refactor the loading and rendering code.

Reply:
I think it's will be enough to keep the have the power-of-two scale function as it should cover our needs. Backgrounds would be fine to be downscaled in a such way. But right, do that at another PR. I'll try to ensure that Wii and 3DS cases aren't broken.

Reply:
Okay, at least, removal of these lines doesn't matter for 3DS and Wii.